### ISSUE TITLE
Default Repo Rules (RFC / discussion)

### ISSUE THREAD

**Issue by DavidBuchanan314** [webosbrew/apps-repo#6]
There's already a short list of rules in the readme: https://github.com/webosbrew/apps-repo/blob/main/README.md

The purpose of this issue is to discuss the rules and/or propose new ones.

I have some proposed additions:

 - Your app must be licensed under a recognized Open Source license.
 
 - Your app must not contain any obfuscated code.
 
Some sources of inspiration:

https://extensionworkshop.com/documentation/publish/add-on-policies/

**Comment by mariotaku** [comment]
I was tend to have permissive rules, such as allowing closed source application. But if we enforce open source rule in central repo, we can certainly lower the risk of providing malicious app.

**Comment by mariotaku** [comment]
Another idea: avoid porting desktop games/applications directly (not a hard requirement)

**Comment by Informatic** [comment]
> Another idea: avoid porting desktop games/applications directly (not a hard requirement)

You mean porting apps without adjusting for different input method / adding "TV-style" UX?

**Comment by mariotaku** [comment]
@Informatic Yes, also porting some apps like QEMU isn't a good idea. I think these are some guidelines instead of rules.

**Comment by DavidBuchanan314** [comment]
> I was tend to have permissive rules, such as allowing closed source application. But if we enforce open source rule in central repo, we can certainly lower the risk of providing malicious app.

Yes. While I don't like restrictive rules in general, we have a certain amount of accountability if buggy/malicious code damages someone's TV. Being picky doesn't cost us anything.

I suspect a less-restricted public repo will pop up at some point, if there is sufficient demand, and I think that's a good thing (as long as people using it are aware of whatever risks they're taking). And of course, individual developers can still host their own repos.

It might be worth noting that the official Ubuntu repos are split into four seperate repos: (https://help.ubuntu.com/community/Repositories/Ubuntu)

> 1. **Main** - Canonical-supported free and open-source software.
> 
> 2. **Universe** - Community-maintained free and open-source software.
> 
> 3. **Restricted** - Proprietary drivers for devices.
> 
> 4. **Multiverse** - Software restricted by copyright or legal issues. 

I think we should aim to be like Ubuntu's "universe" repo, and allow the wider community to provide the rest.
